tests/integration/commands/clone/clone.test.ts

```diff
@@ -0,0 +1,530 @@
+import { sep } from 'node:path'
+
+import { describe, expect, it, beforeEach } from 'vitest'
+import js from 'dedent'
+import stripAnsi from 'strip-ansi'
+
+import { withMockApi } from '../../utils/mock-api.js'
+import { withSiteBuilder } from '../../utils/site-builder.js'
+import { answerWithValue, CONFIRM, DOWN, handleQuestions } from '../../utils/handle-questions.js'
+import { createMock } from '../../utils/mock-execa.js'
+import execa from 'execa'
+import { cliPath } from '../../utils/cli-path.js'
+
+describe.concurrent('clone command', () => {
+  let execaMock: Awaited<ReturnType<typeof createMock>>[0]
+
+  beforeEach(async () => {
+    ;[execaMock] = await createMock(js`
+      module.exports = function execa(command, args) {
+        // Mock git clone command
+        if (command === 'git' && args[0] === 'clone') {
+          const targetDir = args[2]
+          return require('fs/promises').mkdir(targetDir, { recursive: true })
+            .then(() => {
+              const stdout = 'Mocked git clone received: ' + command + ' ' + args.join(' ')
+              console.log(stdout)
+              return { stdout, stderr: '' }
+            })
+        }
+        // For any other command, use the real execa
+        // Normalize paths for Windows
+        const realExeca = require('${require.resolve('execa').split(sep).join('/')}')
+        // execa's APi is... really weird
+        const result = Promise.resolve(realExeca(command, args))
+        result.unref = () => {}
+        return result
+      }
+    `)
+  })
```

Comment:
oof, this took me a while to figure out 😓. Because we run CLI commands via a spawned child process, we can't actually mock anything. This seemingly hasn't come up much before, but here we're interacting with a third-party server (to clone from git remotes), so it's strongly desirable to mock that out. It turns out there was already this helper and pattern for this, but then of course every possible complication turned up here. Hope this is clear enough...